### PR TITLE
Migrate nvtext generate_ngrams APIs to pylibcudf

### PR DESCRIPTION
**Pull request by Matt711**
## Description

Apart of #15162
## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf/cudf/_lib/nvtext/generate_ngrams.pyx [hunk 1]
```diff
@@ -2,75 +2,37 @@
...
-                c_ngrams,
-                c_separator[0]
-            )
-        )
-
-    return Column.from_unique_ptr(move(c_result))
+    result = nvtext.generate_ngrams.generate_ngrams(
+        strings.to_pylibcudf(mode="read"),
+        <size_type> ngrams,
+        <plc_Scalar> py_separator.device_value.c_value
```
Comment (mroeschke) [inline python/cudf/cudf/_lib/nvtext/generate_ngrams.pyx:18]: I think we don't need these explicit casts. I _believe_ Cython will implicitly perform the valid cast if needed

Comment (Matt711) [reply python/cudf/cudf/_lib/nvtext/generate_ngrams.pyx:18]: Thanks, I'll try and build without them. I wonder--in general where you can get away with not explicitly casting?

File: python/pylibcudf/pylibcudf/tests/test_nvtext_generate_ngrams.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        ngram,
+    )
+    expected = pa.array([["ab"], ["cd", "de"], ["fg", "gh"]])
+    if ngram == 3:
+        expected = pa.array([[], ["cde"], ["fgh"]])
+    assert_column_eq(result, expected)
+
+
+@pytest.mark.parametrize("ngram", [2, 3])
+def test_hash_character_ngrams(input_col, ngram):
```
Comment (Matt711) [inline python/pylibcudf/pylibcudf/tests/test_nvtext_generate_ngrams.py:42]: Feel free to suggest a better test here

Comment (bdice) [reply python/pylibcudf/pylibcudf/tests/test_nvtext_generate_ngrams.py:42]: You might be able to generate the ngrams and then call the appropriate cudf hash function on the strings column child of the lists column and reassemble the result, but that may not be significantly better.

Ways to improve this test:
- Assert that the result has the right dtype (probably lists of int32?)
- Write the comparison with respect to the input length and avoid hardcoded values

File: python/pylibcudf/pylibcudf/tests/test_nvtext_generate_ngrams.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    )
+    pa_result = plc.interop.to_arrow(result)
+    if ngram == 2:
+        assert len(pa_result[0]) == 1
+        assert len(pa_result[1]) == 2
+        assert len(pa_result[2]) == 2
+    else:
+        assert len(pa_result[0]) == 0
+        assert len(pa_result[1]) == 1
+        assert len(pa_result[2]) == 1
```
Comment (bdice) [inline python/pylibcudf/pylibcudf/tests/test_nvtext_generate_ngrams.py:55]: Try something like this:
```suggestion
    assert all(len(got) == max(0, len(s) - ngram + 1) for got, s in zip(pa_result, input_col))
```

or

```suggestion
    for got, s in zip(pa_result, input_col):
        expected_length = max(0, len(s) - ngram + 1)
        assert len(got) == expected_length
```

**Review by bdice — APPROVED** [review]
A couple of test suggestions, otherwise LGTM.

File: python/pylibcudf/pylibcudf/tests/test_nvtext_generate_ngrams.py (hunk 3 above)
Comment (Matt711) [reply python/pylibcudf/pylibcudf/tests/test_nvtext_generate_ngrams.py]: Thanks! I went with the first option.

**Pull request merged**